x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/GetStackTracesRequest.java

```diff
@@ -221,6 +219,28 @@ public void parseXContent(XContentParser parser) throws IOException {
         }
     }
 
+    private String[] parseIndices(XContentParser parser) throws IOException {
+        XContentParser.Token token;
+        List<String> indices = new ArrayList<>();
+        while ((token = parser.nextToken()) != XContentParser.Token.END_ARRAY) {
+            if (token == XContentParser.Token.VALUE_STRING) {
+                indices.add(parser.text());
+            } else {
+                throw new ParsingException(
+                    parser.getTokenLocation(),
+                    "Expected ["
+                        + XContentParser.Token.VALUE_STRING
+                        + "] but found ["
+                        + token
+                        + "] in ["
+                        + INDICES_FIELD.getPreferredName()
+                        + "]."
+                );
+            }
+        }
+        return indices.toArray(new String[0]);
```

Comment:
Should the String array be pre-allocated with `indices.size()`?

Reply:
That's actually the preferred style in Java, see also https://github.com/JetBrains/intellij-community/blob/master/java/java-impl/src/inspectionDescriptions/ToArrayCallWithZeroLengthArrayArgument.html for the rationale.